src/adapter/src/catalog/migrate.rs

```diff
@@ -121,7 +121,12 @@ pub(crate) async fn migrate(
 
 // Add the `col_num` to all PG column descriptions.
 //
-// TODO(migration): delete in version v.50 (released in v0.48 + 1 additional
+// Note that this will also populate column and table constraints from the
+// upstream database that were never used, so we must clear them out because
+// they were not used to generate the DDL for the subsources and it is now too
+// late to apply them.
+//
+// TODO(migration): delete in version v.51 (released in v0.49 + 1 additional
```

Comment:
Do you mind if you extend `misc/python/materialize/checks/pg_cdc.py` by addding some `PRMARY KEY/UNIQUE` and `NOT NULL` modifiers to the existing columns of the existing tables and then some `EXPLAIN`-style checks in the `validate` section?

If it is not obvious how to hack that in, please let me know and I will do that for you.

Reply:
Added a somewhat minimal version of this; I was struggling to get the optimizer to choose anything other than the primary index and didn't make much traction trying to look into the index selection code quickly. This will, however, ensure that the default index gets applied and has the appropriate unique constraint applied.